pkg/cmd/server/api/validation/oauth.go

```diff
@@ -18,8 +18,16 @@ import (
 func ValidateOAuthConfig(config *api.OAuthConfig) ValidationResults {
 	validationResults := ValidationResults{}
 
+	if config.MasterCA != nil && len(*config.MasterCA) > 0 {
+		validationResults.AddErrors(ValidateFile(*config.MasterCA, "masterCA")...)
+	} else {
+		validationResults.AddWarnings(fielderrors.NewFieldInvalid("masterCA", "", "connections to the masterURL for OAuth token exchange will use system roots"))
+	}
+
 	if len(config.MasterURL) == 0 {
 		validationResults.AddErrors(fielderrors.NewFieldRequired("masterURL"))
+	} else if _, urlErrs := ValidateURL(config.MasterURL, "masterURL"); len(urlErrs) > 0 {
+		validationResults.AddErrors(urlErrs...)
```

Comment:
this keeps the structure clean if we wanted to chain more checks after this if block for a clean url. it also matches masterPublicURL handling